src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -429,16 +433,24 @@ internal override void AbortRead(long errorCode)
         internal override void AbortWrite(long errorCode)
         {
             ThrowIfDisposed();
+            Debug.Assert(errorCode != 0);
 
             bool shouldComplete = false;
 
             lock (_state)
             {
+                if (!_canWrite)
```

Comment:
I'm confused, some of the code handling `_canRead/Write` is under lock some isn't. What is the source of truth?

Reply:
+1... This reinforces to me that we probably shouldn't be modifying _canRead/Write, and instead just check read/write state as necessary

Reply:
I normally put it under lock if we are changing state. I put it here to keep it with the other state checks.